### PR TITLE
feat: 1295 advanced configuration pt 1

### PR DESCRIPTION
**Pull request by frankvonhoven**
## **Description**

This adds the MVP of the Basic Functionality feature

## **Dev Note**
TODO:
- [ ] Add toast for "Update to Privacy Policy"
- [ ] Get translations
- [x] Add list of api calls
## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2211

## **Manual testing steps**
(Full testing steps from QA available in ticket above)

### Env setup (feature flagged)
➡️ **Add `export BASIC_FUNCTIONALITY=true` to `.js.env`**

* #### Create a new Wallet
  a. Skip SRP
  - [ ] Should go to Onboarding Success Screen --> Default Settings --> Disable Basic Functionality

* #### Create a new Wallet
  a. Complete SRP
  - [ ] Should go to Onboarding Success Screen --> Default Settings --> Disable Basic Functionality

* #### Import a Wallet
  - [ ] Should go to Onboarding Success Screen --> Default Settings --> Disable Basic Functionality

* #### Settings Screen
  a. Open app to Wallet view
  b. Go to Settings > Privacy
  c. Scroll down to "General" > Basic functionality
  - [ ] Should be able to toggle Basic functionality
  - [ ] If OFF, should see Banner Alert on Wallet View

### **New Wallet + SRP**

https://github.com/MetaMask/metamask-mobile/assets/141057783/3c7e907e-c3f6-4596-adf9-905e6a870cd9

### **New Wallet + Skip SRP**

https://github.com/MetaMask/metamask-mobile/assets/141057783/266f7dfe-0636-4bb7-ae60-4402fc810565

### **Import Wallet**

https://github.com/MetaMask/metamask-mobile/assets/141057783/bceb80cc-8eb8-4148-8f9a-fd6cd1c12b40

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/AccountBackupStep1/index.js [hunk 1]
```diff
@@ -191,11 +192,22 @@ const AccountBackupStep1 = (props) => {
...
-      props.navigation.reset({ routes: [{ name: 'HomeNav' }] });
+      process.env.BASIC_FUNCTIONALITY
+        ? goToSuccessFlow()
+        : props.navigation.reset({ routes: [{ name: 'HomeNav' }] });
     } else {
       props.setOnboardingWizardStep(1);
-      props.navigation.reset({ routes: [{ name: 'HomeNav' }] });
+      process.env.BASIC_FUNCTIONALITY
+        ? goToSuccessFlow()
+        : props.navigation.reset({ routes: [{ name: 'HomeNav' }] });
```
Comment (darkwing) [inline app/components/Views/AccountBackupStep1/index.js:210]: What's the difference in these two if/else blocks besides `props.setOnboardingWizardStep(1);`?  Looks like repeated code.

Comment (frankvonhoven) [reply app/components/Views/AccountBackupStep1/index.js:210]: That's the only difference. That logic is what was already there

**Comment by sethkfman** [comment]
@frankvonhoven Option to test the traffic: https://proxyman.io

**Review by MarioAslau — DISMISSED** [review]
Looks Great! Awesome work Frank!

File: app/store/sagas/index.ts [hunk 2]
```diff
@@ -14,6 +14,14 @@ import { Task } from 'redux-saga';
...
 import Logger from '../../util/Logger';
 import LockManagerService from '../../core/LockManagerService';
+import { isObject } from '@metamask/utils';
+import AppConstants from '../../../app/core/AppConstants';
+
+// We are intercepting all fetch request to gate api calls
+// based on privacy settings.
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+const originalFetch = window.fetch; // Never touch this!
```
Comment (NicolasMassart) [inline app/store/sagas/index.ts:24]: Q: could you explain more how this works and why we call a window object that is not expected to even exist here?

File: app/components/Views/Success/index.tsx [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+  const goToDefaultSettings = () => {
+    navigation.navigate(Routes.ONBOARDING.SUCCESS_FLOW, {
+      screen: Routes.ONBOARDING.DEFAULT_SETTINGS,
+    });
+  };
+
+  const handleLink = () => {
+    Linking.openURL(
+      'https://community.metamask.io/t/what-is-a-secret-recovery-phrase-and-how-to-keep-your-crypto-wallet-secure/3440',
```
Comment (NicolasMassart) [inline app/components/Views/Success/index.tsx:121]: Suggestion: should we put htis URL in app/constants/urls.ts ?

**Review by NicolasMassart — DISMISSED** [review]
Great work @frankvonhoven!
I added a comment and a suggestion, but they do not change how the code works, it's more for readability and maintainability.

File: app/components/Views/Success/index.tsx (hunk 3 above)
Comment (frankvonhoven) [reply app/components/Views/Success/index.tsx:121]: Yeah, meant to do that, good catch!

**Comment by mmragkandala** [comment]
**1. Manage default settings icon is not aligned**
The spacing and alignment of the gear icon in the `Manage default settings` seems to be off on both iOS and Android

_Figma_
image

_iOS_
Screenshot 2024-03-28 at 3 46 43 PM

_Android_
Screenshot 2024-03-28 at 3 53 23 PM

**2. Learn more about privacy is missing a link**
On the manage default settings screen during onboarding, the last sentence of the first paragraph should be linked to a support article.

_Figma_
image

_iOS_
Screenshot 2024-03-28 at 3 47 11 PM

_Android_
image

**3. Warning icon in bottom sheet is using wrong color and size**
The warning icon in the turn-off basic functionality bottom sheet is using the wrong color. It is currently using warning/orange but should be error/red. It also appears to be a smaller icon size compared to the designs (XL - 32px)

_Figma_
image

_iOS_
Screenshot 2024-03-28 at 3 50 39 PM

_Android_
Screenshot 2024-03-28 at 3 58 13 PM

File: locales/languages/en.json [hunk 4]
```diff
@@ -415,7 +433,11 @@
     "nfts_autodetect_cta": "Turn on Autodetect NFTs in Settings",
     "turn_on_network_check_cta": "Turn on network details check",
     "display_nft_media_cta_new_1": "To see an NFT, turn on Display NFT media in",
-    "display_nft_media_cta_new_2": "Settings > Security and Privacy."
```
Comment (EtherWizard33) [inline locales/languages/en.json:418]: maybe a more meaningful name for this key could be settings_menu_path_security_privacy

**Comment by frankvonhoven** [comment]
replaced by https://github.com/MetaMask/metamask-mobile/pull/9208

**Pull request closed without merging**